vmware-event-router/internal/processor/knative.go

```diff
@@ -0,0 +1,117 @@
+package processor
+
+import (
+	"context"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/color"
+	config "github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/config/v1alpha1"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/metrics"
+)
+
+const (
+	knativeDefaultRetryInterval = time.Second * 3
+	knativeDefaultMaxRetries    = 3
+)
+
+// knativeProcessor implements the Processor interface
+type knativeProcessor struct {
```

Comment:
This has changed recently (looks like you were developing from an older code base). Please export this struct and return `*KnativeProcessor` in the constructor.

Reply:
Updated the Code.